src/simulation/decision_model_simulation_results.jl

```diff
@@ -90,23 +90,23 @@ get_cached_variables(res::SimulationProblemResults{DecisionModelSimulationResult
 
 get_cached_results(
```

Comment:
I don't understand why passing the type and not the instance matters here.

Reply:
Philosophically, it makes more sense to me because we really aren't using any properties of the instance rather than its type. Practically, sometimes we don't necessarily *have* an instance, like [here](https://github.com/NREL-Sienna/PowerSimulations.jl/pull/1073/files/fbfb1ca3d4f1200b0299de04f1a5ef1d49f41a8f#diff-805b5dec2828ffb16a0f459d5e6c31d3f5bf3691a6e1562db8ad6030e81b2ca0R534).

Reply:
We’ve had this debate endlessly and have tried to use instances when possible because that’s what the Julia manual recommends. I may not be completely following the code, but don’t you have an instance for all cases where the user passed variables/duals/etc to load_results?

Reply:
Yeah, but consider the case where a user has loaded 2 time windows of a dual in their first call to `load_results!` and wants to load 3 time windows of a variable in their second call. Because the "what time periods do we have cached" information (`res.results_timestamps`) is across the whole results instance rather than per-variable/dual/etc., we have to now upgrade to 3 periods of that old dual. This requires fetching what duals we already have cached even if the user doesn't pass any new duals.

An alternative resolution would be allowing different variables/etc. to be cached to different numbers of time windows, and I'd be in support of that, but I was trying to limit the scope of this PR.

Reply:
Could I get a link to that section of the Julia manual? If this is a convention, it seems like one that needs to quite often be broken to avoid awkwardness, like we do with `get_components`.

Reply:
@daniel-thom I think that was a preference for @kdheepak back in the day and not a hard rule

Reply:
Current guidance is captured here https://docs.julialang.org/en/v1/manual/types/#man-singleton-types

Reply:
I can’t remember and can try to look it up. I’m not saying it’s wrong to use types; clearly, we have to do that in get_components. I’m just saying that it you have an instance or can easily construct one, Julia prefers that you use it.